PERFORMANCE.md

```diff
@@ -0,0 +1,59 @@
+# Performance
+
+It is important to be conscious of performance in hot code paths. In such code paths there should be 3 goals:
+
+- Be as fast as possible
+- Be memory efficient
+- Be as stable as possible
+
+This guide is meant to document known performance issues so that the contributors are aware of them.
+
+## Performance Sensitive Areas
+
+- Change detection: `observer`, `computed`
+- View creation: `View`, `Renderer`
+- Routing: `Router`
+- Dependency Injection: `Container`, `Registry`
+
+__Note :__ If you are not in a performance sensitive area feel free to use these constructs. That being said, please do your due diligence to ensure your contribution is indeed not in a hot path.
+
+## Array Methods
+
+Most of the time using array methods e.g. `forEach`, `map`, `filter`, is fine, but in hot paths these methods can become very costly. One would think that these methods would almost be like macros that would compile into more efficient code, but that is not the case. In hot paths these methods can be up to 10X slower than just writing a `for`-loop.
+
+The reason behind the slowness is largely do to the fact that these methods take a closure which allocates memory. In turn this creates GC pressure and potentially creates a polymorphic call site.
```

Comment:
I think this is very good "how sausage is made" explainer. I can add it as a link.